extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaTuningConfig.java

```diff
@@ -87,6 +89,7 @@ public KafkaTuningConfig(
                                     ? DEFAULT_RESET_OFFSET_AUTOMATICALLY
                                     : resetOffsetAutomatically;
     this.segmentWriteOutMediumFactory = segmentWriteOutMediumFactory;
+    this.intermediateHandoffPeriod = intermediateHandoffPeriod == null ? new Period().withDays(Integer.MAX_VALUE) : intermediateHandoffPeriod;
```

Comment:
Does this mean that time-based intermediate handoff is effectively disabled if it is null? What do you think about disabling it explicitly if it's null?

Reply:
Also, please check the formatting.

Reply:
Fixed formatting and yes it means the intermediate handoff is effectively disabled. It didn't wanted to have another flag check so that's why just setting it to a long period. I see any issues with this then I can change.

Reply:
I think, from the user perspective, it looks not clear if they don't want to use this feature. If someone doesn't want to use it, then he/she might be confused what value should be set and should calculate how long period should be set to turn off this feature.
If we support turning it off explicitly in some ways, then it will be more clear for users. I think this is more valuable even though we need to add some more flag checks. Finally, I think it's enough if `null` intermediateHandoffPeriod means disabling time-based intermediate handoff. What do you think?

Reply:
How about just mentioning in the docs that `null` is default value and it means disabled. Internally, the `null` can be translated to a very long `Period` effectively disabling the intermediate handoff, does this sounds good ? Or should I introduce a explicit flag ?

Reply:
I think it would be nice if what the doc says and how the code actually works matches. If you think it's better to keep the current implementation, the doc also should say that the default `intermediateHandoffPeriod` is `Integer.MAX_VALUE` days as it currently does. 

I don't have a strong feeling about this.